apps/web-giddh/src/app/mobile-home/mobile-search-branch/mobile-search-branch.component.ts

```diff
@@ -0,0 +1,160 @@
+    public currentOrganizationType: OrganizationType;
+    /** Holds the active company information */
+    public activeCompany: any;
+
+    /** Subject to unsubscribe from all the subscription */
+    private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
+
+    /** @ignore */
+    constructor(
+        private companyActions: CompanyActions,
+        private generalService: GeneralService,
+        private router: Router,
+        private settingsProfileService: SettingsProfileService,
+        private store: Store<AppState>,
+        private settingsBranchAction: SettingsBranchActions,
+    ) {
+
+    }
+
+    /**
+     * Initializes the component
+     *
+     * @memberof MobileSearchBranchComponent
+     */
+    public ngOnInit(): void {
+        this.currentCompanyBranches$ = this.store.pipe(select(appStore => appStore.settings.branches), takeUntil(this.destroyed$));
+        this.currentCompanyBranches$.subscribe(response => {
+            if (response && response.length) {
+                this.currentCompanyBranches = response;
+            }
+        })
+    }
+
+    /**
+     * Unsubscribes from all the listeners
+     *
+     * @memberof MobileSearchBranchComponent
+     */
+    public ngOnDestroy(): void {
+        this.destroyed$.next(true);
+        this.destroyed$.complete();
+    }
+
+    /**
+     * Switches to branch mode
+     *
+     * @param {string} branchUniqueName Branch uniqueName
+     * @memberof MobileSearchBranchComponent
+     */
+    public switchToBranch(branchUniqueName: string, event: any): void {
+        event.stopPropagation();
+        event.preventDefault();
+        const details = {
+            branchDetails: {
+                uniqueName: branchUniqueName
+            }
+        };
+        this.setOrganizationDetails(OrganizationType.Branch, details);
+        if (!RESTRICTED_BRANCH_ROUTES.includes(this.router.url)) {
+            window.location.reload();
+        } else {
+            window.location.href = '/pages/home';
+        }
+    }
+
+    /**
+     * This will get the current company data
+     *
+     * @memberof MobileSearchBranchComponent
+     */
+    public getCurrentCompanyData(): void {
+        this.settingsProfileService.GetProfileInfo().subscribe((response: any) => {
+            if (response && response.status === "success" && response.body) {
+                this.activeCompany = response.body;
+            }
+        });
+    }
+
+    /**
+     * Filters the branches based on text provided
+     *
+     * @param {string} branchName Branch name query entered by the user
+     * @memberof MobileSearchBranchComponent
+     */
+    public filterBranch(branchName: string): void {
+        let branches = [];
+        this.currentCompanyBranches$.pipe(take(1)).subscribe(response => {
+            branches = response || [];
+        });
+        if (branchName) {
+            this.currentCompanyBranches = branches.filter(branch => {
+                if (!branch.alias) {
```

Comment:
nullity check for alias

Reply:
Not required as there is a else block which executes only when alias is present